lightning-invoice/src/lib.rs

```diff
@@ -904,6 +906,79 @@ impl Deref for PositiveTimestamp {
 }
 
 impl Invoice {
+	/// Utility to construct an invoice.
+	pub fn from_channelmanager<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>(
+		channelmanager: &ChannelManager<Signer, M, T, K, F, L>, amt_msat: Option<u64>, description: String,
+		network: Network,
+	) -> Result<Invoice, CreationError>
+	where
+		M::Target: chain::Watch<Signer>,
+		T::Target: BroadcasterInterface,
+		K::Target: KeysInterface<Signer = Signer>,
+		F::Target: FeeEstimator,
+		L::Target: Logger,
+	{
+		// Marshall route hints.
+		let our_channels = channelmanager.list_usable_channels();
+		let mut route_hints = vec![];
+		let mut min_final_cltv_expiry = 18;
```

Comment:
Maybe use the new constant in one of the later updates of the dependent PRs https://github.com/rust-bitcoin/rust-lightning/pull/893/commits/615ef7d6f88464ed8478a8c8cc0e7582ba66d773